src/test/java/com/quorum/gauge/BlockSynchronization.java

```diff
@@ -166,21 +167,45 @@ public void proposeValidator(Node targetNode, List<Node> nodes) {
     }
 
     public void proposeValidatorImpl(Node targetNode, List<Node> nodes, boolean vote) {
-        IstanbulNodeAddress nodeAddressRes = istanbulService.nodeAddress(QuorumNode.valueOf(targetNode.getName())).blockingFirst();
+        IstanbulNodeAddress nodeAddressRes = getNodeAddress(targetNode);
         Response.Error err1 = Optional.ofNullable(nodeAddressRes.getError()).orElse(new Response.Error());
         assertThat(err1.getMessage()).as("istanbul.nodeAddress must succeed").isBlank();
         String nodeAddr = nodeAddressRes.getResult();
         logger.debug("node {} node address: {}", targetNode.getName(), nodeAddr);
         nodes.stream().forEach(n -> {
             logger.debug("istanbul.propose targetNode:{} fromNode:{} vote:{} nodeAddr:{}", targetNode.getName(), n.getName(), vote, nodeAddr);
-            istanbulService.propose(QuorumNode.valueOf(n.getName()), nodeAddr, vote)
+            propose(n, nodeAddr, vote)
                 .doOnNext(res -> {
                     Response.Error err2 = Optional.ofNullable(res.getError()).orElse(new Response.Error());
                     assertThat(err2.getMessage()).as("istanbul.propose must succeed").isBlank();
                 }).blockingSubscribe();
         });
     }
 
+    public IstanbulNodeAddress getNodeAddress(Node node) {
+        if(isBesuNode(node)) {
+            return besuService.nodeAddress(QuorumNode.valueOf(node.getName())).blockingFirst();
+        }
+        return istanbulService.nodeAddress(QuorumNode.valueOf(node.getName())).blockingFirst();
```

Comment:
should the `istanbulService.nodeAddress(...)` function in the `istanbulService` do this `if`? so that will be separate better the concerns. same applies for the other lines

Reply:
Done. Pushed a commit